source/gameengine/GameLogic/Joystick/SCA_JoystickEvents.cpp

```diff
@@ -103,29 +83,17 @@ void SCA_Joystick::HandleEvents(void)
 		 * it wont work as it should */
 		
 		switch (sdl_event.type) {
-			case SDL_JOYAXISMOTION:
-				SCA_Joystick::m_instance[sdl_event.jaxis.which]->OnAxisMotion(&sdl_event);
-				break;
-			case SDL_JOYHATMOTION:
-				SCA_Joystick::m_instance[sdl_event.jhat.which]->OnHatMotion(&sdl_event);
-				break;
-			case SDL_JOYBUTTONUP:
-				SCA_Joystick::m_instance[sdl_event.jbutton.which]->OnButtonUp(&sdl_event);
-				break;
-			case SDL_JOYBUTTONDOWN:
-				SCA_Joystick::m_instance[sdl_event.jbutton.which]->OnButtonDown(&sdl_event);
+			case SDL_CONTROLLERDEVICEADDED:
+			case SDL_CONTROLLERDEVICEREMOVED:
+				/* pass */
 				break;
-#if 0	/* Not used yet */
-			case SDL_JOYBALLMOTION:
-				SCA_Joystick::m_instance[sdl_event.jball.which]->OnBallMotion(&sdl_event);
+			case SDL_CONTROLLERBUTTONDOWN:
+			case SDL_CONTROLLERBUTTONUP:
+				SCA_Joystick::m_instance[sdl_event.cbutton.which]->OnButtonEvent(&sdl_event);
 				break;
-#endif
-#if SDL_VERSION_ATLEAST(2, 0, 0)
-			case SDL_JOYDEVICEADDED:
-			case SDL_JOYDEVICEREMOVED:
-				/* pass */
+			case SDL_CONTROLLERAXISMOTION:
```

Comment:
Why using all CONTROLLER functions and not JOYSTICK ones ?
My right axis isn't detected here but if you replace by SDL_JOYAXISMOTION it hlaf works.